src/modules/menu/components/share/MenuShareEmbed.vue

```diff
@@ -120,6 +117,18 @@ function toggleEmbedSharing() {
 
 function togglePreviewModal() {
     showPreviewModal.value = !showPreviewModal.value
+    if (showPreviewModal.value) {
+        window.addEventListener('message', onPreviewChange)
+    } else {
+        window.removeEventListener('message', onPreviewChange)
+    }
+}
+
+function onPreviewChange(e) {
+    if (e?.data?.type === 'gaChange') {
```

Comment:
Might be a better idea to import the `gaChange` string from the iframe api file here

Reply:
In the api file we should have also each message type as variable and documented same as their payload.